### PR TITLE
15915 updated limited restoration extension expiry date

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#15915

*Description of changes:*

Fix limitedRestorationExtension.expiry_date duration (3 -> 2 years)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/restoration.py [hunk 1]
```diff
@@ -68,15 +68,16 @@ def validate(business: Business, restoration: Dict) -> Optional[Error]:
...
-    # Between 1 month and 3 years in the future for limited restoration extension
     msg = []
     expiry_date_path = '/filing/restoration/expiry'
+
     if expiry_date := get_date(filing, expiry_date_path):
-        now = LegislationDatetime.now().date()
-        max_expiry_years = 2 if restoration_type == 'limitedRestoration' else 3
+        max_expiry_years = 2
+        now = LegislationDatetime.as_legislation_timezone(business.restoration_expiry_date).date()
+        if restoration_type == 'limitedRestoration': now = LegislationDatetime.now().date()
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/restoration.py:79]: `restoration_expiry_date` will be null while filing a `limitedRestoration`.
```suggestion
        now = LegislationDatetime.now().date()
        if restoration_type == 'limitedRestorationExtension': 
            now = LegislationDatetime.as_legislation_timezone(business.restoration_expiry_date).date()
```

File: legal-api/tests/unit/services/filings/validations/test_restoration.py [hunk 2]
```diff
@@ -197,16 +197,16 @@ def test_validate_relationship(session, test_status, restoration_type, expected_
...
+        ('invalid_greater', 'limitedRestorationExtension', relativedelta(years=2, days=1), False),
         ('lesser', 'limitedRestorationExtension', relativedelta(months=1), True),
         ('invalid_lesser', 'limitedRestorationExtension', relativedelta(days=25), False),
         ('missing', 'limitedRestorationExtension', None, False),
     ]
 )
 def test_validate_expiry_date(session, test_name, restoration_type, delta_date, is_valid):
     """Assert that expiry date is validated."""
-    business = Business(identifier='BC1234567', legal_type='BC')
+    business = Business(identifier='BC1234567', legal_type='BC', restoration_expiry_date=LegislationDatetime.now())
```
Comment (vysakh-menon-aot) [inline legal-api/tests/unit/services/filings/validations/test_restoration.py:209]: Add test to consider above comment (by setting `restoration_expiry_date` to None when `restoration_type` is `limitedRestoration`)

**Review by vysakh-menon-aot — CHANGES_REQUESTED** [review]

**Review by argush3 — APPROVED** [review]
